Fix print_buffer

zoechi opened the pull request:
It's still not perfect, but I think it's pretty useful already. A nuisance is that the test output starts immediately after the last character in the last line of the widget.
I haven't found a way to prevent that.

zoechi committed:
Fix print_buffer

It's still not perfect, but I think it's pretty useful already.
A nuisance is that the test output starts immediately after the last character
in the last line of the widget.
I haven't found a way to prevent that.

Philipp-M commented on src/test_helper.rs:
```diff
@@ -208,26 +205,24 @@ impl Widget for DebugWidget {
+
+        terminal.clear()?;
         terminal.current_buffer_mut().clone_from(buffer);
         terminal.flush()?;
```
You can add the following to get to the beginning of the next line:

```suggestion
        terminal.flush()?;
        crossterm::queue!(stdout(), crossterm::cursor::MoveTo(0, buffer.area.height))?;
```

zoechi replied:
Sounds good. I'll try.

zoechi committed:
Ensure errors from the spawned thread are not masked

fixes #13

The error with multiple render_view() tests was caused by global logging being
initialized more than once.
See also https://github.com/tokio-rs/console/issues/505

My workaround is a bit hacky, but I think there are a bunch of things, including
logging, that should be set up using a configuation struct passed to `fn new()`
anyway. Cleaning that workaround up would go with that task automatically.

zoechi committed:
Update src/test_helper.rs

Co-authored-by: Philipp Mildenberger <philipp@mildenberger.me>

zoechi committed:
Merge commit '8b836ea' into print_buffer_test_helper

Philipp-M reviewed (approved):
Thanks for the fix. LGTM :)
